src/stores/pacbioPoolCreate.js

```diff
@@ -246,29 +249,27 @@ export const usePacbioPoolCreateStore = defineStore('pacbioPoolCreate', {
     },
     /**
      * Returns a list of requests.
-     * If IDs are provided, returns the requests with those IDs and a selected property indicating if they are in the selected requests.
-     * Otherwise, returns all requests with a selected property indicating if they are in the selected requests.
+     * If a source object is provided, returns the requests for that source object.
+     * Otherwise, returns all requests with a selected property indicating if they are in the selected requests with their source_id.
      *
      * @param {Object} state - The Pinia state object.
-     * @returns {Function} A function that takes an array of IDs and returns the corresponding requests with a selected property.
+     * @returns {Function} A function that takes a source object and returns the corresponding requests with their source_id.
      *
-     * @example
-     * // Get all requests
-     * const allRequests = requestList(state)();
-     * // Get specific requests
-     * const specificRequests = requestList(state)(['id1', 'id2']);
      */
-    requestList: (state) => (ids) => {
+    requestList: (state) => (source_obj) => {
       const requests = state.resources.requests
-      const selectedRequests = state.used_aliquots
+      const selectedUsedAliquots = state.used_aliquots
       let val = []
-      if (ids) {
-        val = ids.map((id) => {
-          return { ...requests[id], selected: !!selectedRequests[`_${id}`] }
+      if (source_obj) {
+        const source_id = source_obj.source_id ?? source_obj.id
+        const used_aliquot = state.used_aliquots[`_${source_id}`]
+        const selected = !!used_aliquot
+        val = source_obj['requests']?.map((id) => {
+          return { ...requests[id], selected, source_id }
         })
       } else {
         val = Object.values(requests).map((request) => {
-          return { ...request, selected: !!selectedRequests[`_${request.id}`] }
+          return { ...request, selected: !!selectedUsedAliquots[`_${request.source}`] }
```

Comment:
Is this suppose to be `!!selectedUsedAlqiuots['_${request.source_id}']`?

Reply:
I think it is kept as 'source' inside the request object.I will double check it.

Reply:
It is returned as 'well' or 'tube'  inside request from service.